x-pack/plugin/core/src/main/java/org/elasticsearch/license/LicenseOverrides.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.license;
+
+import org.elasticsearch.common.hash.MessageDigests;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Contains license expiration date overrides.
+ */
+class LicenseOverrides {
+    private LicenseOverrides() {}
+
+    // This approach of just having everything in a hardcoded map is based on an assumption that we will need to do this very infrequently.
+    // If this assumption proves incorrect, we should consider switching to another approach,
+    private static final Map<String, Date> LICENSE_OVERRIDES;
+
+    static {
+        // This value is not a "real" license ID, it is used for testing this code
+        String TEST_LICENSE_ID_HASH = MessageDigests.toHexString(
+            MessageDigests.sha256().digest("20f0e481-0bfa-4f63-938c-259a3fc6ec01".getBytes(StandardCharsets.UTF_8))
```

Comment:
Where did this test UUID come from? Are we sure it's not in use somewhere?
Did we consider using something more test-like for testing (e.g. `"12345678-abcd-0000-0000-000000000000"`)

Reply:
Whoop, I meant to do that - this UUID was one that I generated a license with. Good catch, fixed!

(I'm honestly more worried about collisions in "fakey" UUIDs than actual random ones, given the properties of random ones, but I agree using one that's fakey here is a good idea)